lib/paramedic.js

```diff
@@ -106,14 +106,21 @@ ParamedicRunner.prototype = {
         }).then(function(res) {
             if (self.testsPassed) {
                 logger.info("All tests have been passed.");
-                process.exit(0);
+                return 0;
             } else {
                 logger.error("There are tests failures.");
-                process.exit(1);
+                return 1;
             }
         }, function(err) {
             logger.error("Failed: " + err);
-            process.exit(2);
+            return 2;
+        }).then(function(exitCode){
+            if(self.config.isCleanUpAfterRun()) {
+                logger.info("cordova-paramedic: Deleting the application: " + self.tempFolder.name);
+                shell.cd("..");
```

Comment:
@dblotsky Refer line [135](https://github.com/sarangan12/cordova-paramedic/blob/Cleanup_Application/lib/paramedic.js#L135). In that line, the shell enters the temporary application directory created and then executes all the commands (platform install, plugin install, run, etc.) So, at the end, if I execute "rm -rf" command, the command will remove all files and sub directories - but it will not delete the application directory (will give EBUSY error). So, we are moving to the parent directory (in essence reversing the line [135](https://github.com/sarangan12/cordova-paramedic/blob/Cleanup_Application/lib/paramedic.js#L135)) and deleting the entire directory.